cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs

```diff
@@ -156,119 +154,147 @@ impl Worker for OperatorWorker {
+                let cipher = eigen_crypto::ec::suite_b::ecies::encrypt(
+                    &public_key,
+                    &s1,
+                    &s2,
+                    &result.to_bytes_be(),
+                )
+                .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+
+                // 5. Result is cipher, encode it as base64
+                let result = base64::encode(&cipher);
                 Ok(result)
             }
             OperatorKind::Encrypt => {
                 // NOTE: `input.operand_1` is acually plain text
                 let op1 = BigUint::parse_bytes(input.operand_1.as_bytes(), 10)
                     .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
                 let op_bytes = op1.to_bytes_be();
-                // TODO: First, do ECIES encrypt
 
-                let cipher = op_bytes;
-
-                // Second, do AES encrypt
-                let aes_key = register_func::get_aes_key();
+                // 1. Do ECIES encrypt
+                let s1 = vec![];
+                let s2 = vec![];
+                let key_pair = register_func::get_key_pair();
+                let alg = &eigen_crypto::sign::ecdsa::ECDSA_P256_SHA256_ASN1;
+                let public_key =
+                    eigen_crypto::sign::ecdsa::UnparsedPublicKey::new(alg, key_pair.public_key());
                 let cipher =
-                    eigen_crypto::ec::suite_b::ecies::aes_encrypt_less_safe(aes_key, &cipher)
+                    eigen_crypto::ec::suite_b::ecies::encrypt(&public_key, &s1, &s2, &op_bytes)
                         .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
-                let b = BigUint::from_bytes_be(&cipher[..]);
-                let result = b.to_str_radix(10);
+
+                // 2. Result is cipher, encode it as base64
+                error!("[ENCRYPT] Cipher: {:?}", cipher);
+                let result = base64::encode(&cipher);
+                error!("[ENCRYPT] Cipher Base64: {:?}", result);
                 Ok(result)
             }
             OperatorKind::Decrypt => {
-                let b = BigUint::parse_bytes(input.operand_1.as_bytes(), 10)
-                    .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
-                let cipher = b.to_bytes_be();
-
-                // First, do AES decrypt
-                let aes_key = register_func::get_aes_key();
-                let operand_1 =
-                    eigen_crypto::ec::suite_b::ecies::aes_decrypt_less_safe(aes_key, &cipher)
+                // 1. Cipher is encoded as base64, should be decoded
+                error!("[DECRYPT] Cipher Base64: {:?}", input.operand_1);
+                let cipher_operand = base64::decode(&input.operand_1)
+                    .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
+                error!("[DECRYPT] Cipher: {:?}", cipher_operand);
```

Comment:
usless log.

Reply:
Should be removed here. It is just used for debugging.